pyscf/gto/mole.py

```diff
@@ -4071,6 +4071,48 @@ def fakemol_for_charges(coords, expnt=1e16):
     fakemol._built = True
     return fakemol
 
+def fakemol_for_cgtf_charge(coord, expnt=1e16, contr_coeff=1):
+    '''Construct a fake Mole object that holds the charge on the given
+    coordinate (coord).  The shape of the charge can be a linear
+    combination of gaussian type functions with exponents expnt and
+    contraction coefficients contr_coeff.
+    '''
+    nbas = coord.shape[0]
+    expnt = numpy.asarray(expnt).ravel()
+    contr_coeff = numpy.asarray(contr_coeff).ravel()
+    
+    fakeatm = numpy.zeros((nbas,ATM_SLOTS), dtype=numpy.int32)
+    fakebas = numpy.zeros((nbas,BAS_SLOTS), dtype=numpy.int32)
+    fakeenv = [0] * PTR_ENV_START
+    ptr = PTR_ENV_START
+    fakeatm[:,PTR_COORD] = numpy.arange(ptr, ptr+nbas*3, 3)
+    fakeenv.append(coord.ravel())
+    ptr += nbas*3
+    fakebas[:,ATOM_OF] = numpy.arange(nbas)
+    fakebas[:,NPRIM_OF] = contr_coeff.size
+    fakebas[:,NCTR_OF] = 1
+    if expnt.size == 1:
+        expnt = expnt[0]
+        # approximate point charge with gaussian distribution
+        # contr_coeff*exp(-expnt*r^2)
+        fakebas[:,PTR_EXP] = ptr
+        fakebas[:,PTR_COEFF] = ptr+1
+        fakeenv.append([expnt, 1 / (2*numpy.sqrt(numpy.pi)*gaussian_int(2,expnt))])
+        ptr += 2
+    else:
+        assert expnt.size == nbas or expnt.size == contr_coeff.size
```

Comment:
```suggestion
        assert expnt.size == contr_coeff.size
```

The `expnt.size == nbas` comes from `fakemol_for_charges` where it has a different effect: it looks like you can feed that routine a set of coordinates with a corresponding set of exponents.

Reply:
Does this routine work correctly if you feed many coordinates? The function should place the same exponents and contraction coefficients on every one of those coordinates. In practice, one could do this to calculate the potential of H2O, for instance: place hydrogen atom potentials on the two hydrogens.